src/unix/lwt_unix_unix.h

```diff
@@ -1860,7 +1851,7 @@ static value result_readdir_n(struct job_readdir_n *job)
         result = caml_alloc(job->count, 0);
         long i;
         for (i = 0; i < job->count; i++) {
-            Store_field(result, i, caml_copy_string(job->entries[i].d_name));
+            Store_field(result, i, caml_copy_string(job->entries[i]));
```

Comment:
We should also free the `entries` that actually got allocated, both if the job fails and if it succeeds. For that, we need a valid `job->count` in all cases, but we don't have one in case of an error from `readdir` in the worker (because of the hard `return` in it, above).